src/components/bottomSheet/BottomSheet.tsx

```diff
@@ -659,7 +659,12 @@ const BottomSheetComponent = forwardRef<BottomSheet, BottomSheetProps>(
           status === KEYBOARD_STATUS.SHOWN &&
           position !== closedDetentPosition
         ) {
-          index = highestDetentPosition ?? DEFAULT_KEYBOARD_INDEX;
+          if (detents !== undefined && highestDetentPosition !== undefined) {
+            index = detents.indexOf(highestDetentPosition);
+          }
+          if (index === -1) {
+            index = DEFAULT_KEYBOARD_INDEX;
+          }
```

Comment:
I kept this "default index" fallback, but tbh I'm not sure what is its purpose.
I'd rather have `-1` instead of a "random" number if things go bad.